drivers/sensor/bmi160/bmi160.c

```diff
@@ -712,20 +715,31 @@ static int bmi160_sample_fetch(const struct device *dev,
 	struct bmi160_data *data = dev->data;
 	uint8_t status;
 	size_t i;
+	int ret = 0;
+	enum pm_device_state pm_state;
+
+	(void)pm_device_state_get(dev, &pm_state);
```

Comment:
not familiar with this driver but why don't don't you resume (with `pm_device_runtime_get`) do the operation and suspend (or just release the reference with `pm_device_runtime_put`) ? Is it because of being time sensitive ?